lib/request.js

```diff
@@ -42,11 +42,11 @@ const request = (options, success, error) => {
       res.on('end', () => {
         const parsed = JSON.parse(data);
         if (statusCode >= 200 && statusCode < 300 && typeof success === 'function') {
-          success(parsed);
+          success(parsed, statusCode);
         } else if (typeof error === 'function') {
           const e = new Error(parsed.description);
           e.name = parsed.error;
-          error(e);
+          error(e, statusCode);
```

Comment:
Could you pls give me an example, when do you need to use the `statusCode`? As I could see the documentation, in case of success, any additional info will be available in response's body. In case of error, every error object description contains `error`-field with the code inside.

Reply:
I do it to be in consistent with `success` callback

Reply:
@fan-tom from the error-handling perspective it's not necessary to pass the `statusCode` as it already exists in error-object, which is being passed to error callback-function. So it looks like a data-duplication.

From the perspective of successful cases handling, I'm ready to agree, as I can see that it's much-much easy to distinguish successful cases (i.e. ["File or empty folder"](https://tech.yandex.com/disk/api/reference/move-docpage/#response__file) and ["Non-empty folder"](https://tech.yandex.com/disk/api/reference/move-docpage/#response__folder)) by `statusCode`. But I'm not sure that changing library API is an appropriate way for that. What do you think if we'll make `statusCode` a "private"-property of the `parsed`-object, i.e.:
```js
{
  href: "https://cloud-api.yandex.net/v1/disk/resources?path=disk%3A%2Fbar%2Fphoto.png",
  method: "GET",
  templated: false,
  _statusCode: 201
}
```
?

Reply:
> But I'm not sure that changing library API is an appropriate way for that

AFAIK in JS adding extra parameters to callbacks is backward-compatible change, existing code will continue to work as expected, so I think it is not It's not so bad to pass additional info to callbacks. OTOH, adding extra properties to response data ("private"-property, as you named it) looks like a crutch, it is not forward-compatible (It is unlikely that Yandex will add the same property in the future, but nonetheless). Moreover, it will be very confusing when debugging where this field is originated from, as in yandex api docs it is not present, so in direct calls via curl for example. I think insted of passing status code directly we may pass our own discriminant value, like 'performed'/'processing' for that particular case of clash.
What about passing status code to error callback, I think I can agree with you, I'll remove that changes